amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
@marcoceppi my use case is just the one mentioned in the commit log - that is, we have MAAS cluster with multiple architectures, but we'd like to be able to limit the test to running across a single architecture to e.g. verify arm64 compatibility.